### PR TITLE
[AKS] Be able to create AKS with virtual node enabled

### PR DESCRIPTION
**Pull request by gossion**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Review by levimm — APPROVED** [review]

File: src/aks-preview/azext_aks_preview/_help.py [hunk 1]
```diff
@@ -162,7 +162,8 @@
                 http_application_routing  - configure ingress with automatic public DNS name creation.
                 monitoring                - turn on Log Analytics monitoring. Uses the Log Analytics Default Workspace if it exists, else creates one. Specify "--workspace-resource-id" to use an existing workspace.
                                             If monitoring addon is enabled --no-wait argument will have no effect
-                virtual-node              - enable AKS Virtual Node. Requires --subnet-name to provide the name of an existing subnet for the Virtual Node to use.
+                virtual-node              - enable AKS Virtual Node. Requires --aci-subnet-name to provide the name of an existing subnet for the Virtual Node to use.
+                                            aci-subnet-name must be in the same vent which is specified by --vnet-subnet-id (required as well).
```
Comment (feiskyer) [inline src/aks-preview/azext_aks_preview/_help.py:166]: s/vent/vnet/

File: src/aks-preview/azext_aks_preview/custom.py [hunk 2]
```diff
@@ -1954,6 +1959,16 @@ def _handle_addons_args(cmd,  # pylint: disable=too-many-statements
             addon_profile.config[CONST_ACC_SGX_QUOTE_HELPER_ENABLED] = "false"
         addon_profiles[CONST_CONFCOM_ADDON_NAME] = addon_profile
         addons.remove('confcom')
+    if 'virtual-node' in addons:
+        if not aci_subnet_name or not vnet_subnet_id:
+            raise CLIError('"--enable-addons virtual-node" requires "--aci-subnet-name" and "--vnet-subnet-id".')
+        # TODO: how about aciConnectorwindows, what is its addon name?
```
Comment (feiskyer) [inline src/aks-preview/azext_aks_preview/custom.py:1965]: yes, it should be in same name format

**Review by feiskyer — APPROVED** [review]

**Comment by gossion** [comment]
```
urllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
```
this error is unknown for me. @fengzhou-msft  can you help to re-trigger the test?

**Review by fengzhou-msft — APPROVED** [review]

**Comment by feiskyer** [comment]
@fengzhou-msft thanks for getting this merged.

**Pull request merged**